homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,179 @@
+"""Home Assistant Switcher Component Switch platform."""
+
+from logging import getLogger
+from typing import Any, Callable, cast, Dict, Optional
+
+from homeassistant.components.switch import ATTR_CURRENT_POWER_W, SwitchDevice
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.exceptions import PlatformNotReady
+
+from . import (
+    ATTR_AUTO_OFF_SET, ATTR_DEVICE_NAME, ATTR_ELECTRIC_CURRNET,
+    ATTR_REMAINING_TIME, DATA_DEVICE, DOMAIN, SIGNAL_SWITCHER_DEVICE_UPDATE)
+
+_LOGGER = getLogger(__name__)
+
+DEPENDENCIES = ['switcher_kis']
+
+PROPERTIES_TO_DEVICE_ATTRIBUTES = {
+    'current_power_w': ATTR_CURRENT_POWER_W,
+    'electric_current': ATTR_ELECTRIC_CURRNET,
+    'remaining_time': ATTR_REMAINING_TIME,
+    'auto_off_set': ATTR_AUTO_OFF_SET,
+    'device_name': ATTR_DEVICE_NAME
+}
+
+
+async def async_setup_platform(hass: HomeAssistantType, config: Dict,
+                               async_add_entities: Callable,
+                               discovery_info: Dict) -> None:
+    """Set up the switcher platform for the switch component."""
+    if DOMAIN not in hass.data:
```

Comment:
This should not happen, I would prefer just `assert DOMAIN in Haas.data`

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/e46d9334931a1c263a3caf079f6f8eaf27f3d8bd).